src/transparency/glue.rs

```diff
@@ -303,15 +306,15 @@ where
 impl<F, B> Future for HyperServerSvcFuture<F>
 where
     F: Future<Item=http::Response<B>>,
-    F::Error: fmt::Debug,
+    F::Error: Error + fmt::Debug + Send + Sync + 'static,
 {
     type Item = http::Response<BodyPayload<B>>;
     type Error = h2::Error;
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         let mut res = try_ready!(self.inner.poll().map_err(|e| {
-            debug!("h2 error: {:?}", e);
-            h2::Error::from(io::Error::from(io::ErrorKind::Other))
+            debug!("HTTP/1 error: {}", e);
+            h2::Error::from(io::Error::new(io::ErrorKind::Other, e))
```

Comment:
Do we need a TODO or something about clarifying this? It's strange to construct an `h2::Error` for a HTTP/1 error.

Reply:
Actually, I just checked, and it turns out there's no longer any reason for this to be an `h2::Error` --- the future's error type can now be changed to `S::Error`, and this conversion can be removed.